### PR TITLE
Code optimized 

### PR DESCRIPTION
**Pull request by a2alim**
Code optimization done, 
- Remove One time used Enum
- Update shurjoPay status

### CODE REVIEW

File: src/main/java/com/shurjomukhi/Shurjopay.java [hunk 1]
```diff
@@ -293,52 +217,43 @@ public PaymentReq getDefaultInfo(PaymentReq paymentReq) {
 			paymentReq.setClientIp(InetAddress.getLocalHost().getHostAddress());
 			
 		} catch (UnknownHostException e) {
-			
 			log.warn("Client IP does not found. Setting default ip address..", e);
-			paymentReq.setClientIp(DEFAULT_IP);
+			paymentReq.setClientIp("127.0.0.1");
```
Comment (imtiaz-rahi) [inline src/main/java/com/shurjomukhi/Shurjopay.java]: why fixed client ip ?? not acceptable

File: src/main/java/com/shurjomukhi/Shurjopay.java [hunk 2]
```diff
@@ -175,7 +133,7 @@ public VerifiedPayment verifyPayment(String spTxnId) throws ShurjopayException {
 			verifiedPaymentReq.put("order_id", spTxnId);
 
 			String requestBody = prepareReqBody(verifiedPaymentReq);
-			HttpRequest request = postRequest(requestBody, Endpoints.VERIFIED_PMNT.title(), true);
+			HttpRequest request = postRequest(requestBody, "verification", true);
```
Comment (imtiaz-rahi) [inline src/main/java/com/shurjomukhi/Shurjopay.java]: API URL endpoint is not understandable from this code. Easy to make mistake in future.
Also, any URL path should start with '/'.

File: src/main/java/com/shurjomukhi/ShurjopayStatus.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
+package com.shurjomukhi;
+
+/**
+ * All shurjopay's reserved codes and messages.
```
Comment (imtiaz-rahi) [inline src/main/java/com/shurjomukhi/ShurjopayStatus.java]: Status code, details of shurjopay system defined as ENUM constants

File: src/main/java/com/shurjomukhi/model/Transaction.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import com.shurjomukhi.ShurjopayStatus;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * After verifying payment response
```
Comment (imtiaz-rahi) [inline src/main/java/com/shurjomukhi/model/Transaction.java]: why this new model ?

File: src/test/java/com/shurjopay/ShurjopayTest.java [hunk 5]
```diff
@@ -165,13 +159,15 @@ private void fillupAndSubmitPaymentForm(String url, boolean shouldFail) throws I
...
 	
 	/**
-	 * Prepares chrome driver instance.
+	 * Prepares chrome driver instance
+	 *
 	 * @return prepared chrome driver.
 	 */
 	private WebDriver getChrome() {
-		System.setProperty("webdriver.chrome.driver", "src/test/resources/drivers/chromedriver");
+		System.setProperty("webdriver.chrome.driver", "src/test/resources/drivers/chromedriver.exe");
```
Comment (imtiaz-rahi) [inline src/test/java/com/shurjopay/ShurjopayTest.java]: exe is hardcoded here. what will now happen for Linux or macos ????

**Review by imtiaz-rahi — CHANGES_REQUESTED** [review]
please look at the feedbacks and provide solutions for those

File: src/main/java/com/shurjomukhi/Shurjopay.java (hunk 1 above)
Comment (a2alim) [reply src/main/java/com/shurjomukhi/Shurjopay.java]: Set client IP dynamically that's get from InetAddress. If face exception then handle it by hardcoded. Now clientIp default value set form payment req model class.

File: src/main/java/com/shurjomukhi/Shurjopay.java (hunk 2 above)
Comment (a2alim) [reply src/main/java/com/shurjomukhi/Shurjopay.java]: Fixed it with constructor level.

File: src/main/java/com/shurjomukhi/ShurjopayStatus.java (hunk 3 above)
Comment (a2alim) [reply src/main/java/com/shurjomukhi/ShurjopayStatus.java]: Documentation comment updated

File: src/main/java/com/shurjomukhi/model/Transaction.java (hunk 4 above)
Comment (a2alim) [reply src/main/java/com/shurjomukhi/model/Transaction.java]: This new model came from "[a5108aa]" this commit. No need this model. Now it deleted.

File: src/test/java/com/shurjopay/ShurjopayTest.java (hunk 5 above)
Comment (a2alim) [reply src/test/java/com/shurjopay/ShurjopayTest.java]: Now chrome driver loaded dynamically based on user OS.

File: src/main/java/com/shurjomukhi/Shurjopay.java [hunk 6]
```diff
@@ -291,54 +226,43 @@ public PaymentReq getDefaultInfo(PaymentReq paymentReq) {
...
+	 * @param isAuthHead true if needed "Authorization" header otherwise false
 	 * @return HttpRequest successful http request.
 	 */
-	private HttpRequest postRequest(String httpBody, String endPoint, boolean isAuthHead) {
-		
-		return HttpRequest.newBuilder(URI.create(spConfig.getApiBaseUrl().concat(endPoint)))
-						  .header(AUTH_KEYWORD, getFormattedToken(authToken.getToken(), authToken.getTokenType()))
-						  .POST(HttpRequest.BodyPublishers.ofString(httpBody)).header("Content-Type", "application/json").build();
+	private HttpRequest postRequest(String httpBody, String url, boolean isAuthHead) {
+		var request = HttpRequest.newBuilder(URI.create(url));
```
Comment (imtiaz-rahi) [inline src/main/java/com/shurjomukhi/Shurjopay.java]: private HttpRequest postRequest(String httpBody, String url, boolean isAuthHead) {
		var builder = HttpRequest.newBuilder(URI.create(url))
				.POST(HttpRequest.BodyPublishers.ofString(httpBody))
				.header("Content-Type", "application/json");
		if (isAuthHead) builder.header("Authorization", authToken.getFormatted());
		return builder.build();
	}

File: src/main/java/com/shurjomukhi/model/ShurjopayToken.java [hunk 7]
```diff
@@ -18,36 +18,33 @@
...
 	@JsonProperty("expires_in")
 	private Integer expiredTimeInSecond;
 
-	/** Merchant's registered store id on shurjopay system.*/
+	/** Merchant's registered store id on shurjopay system. */
 	@JsonProperty("store_id")
 	private Integer storeId;
 
-	/** Shurjopay status message. See more.. {@link ShurjopayStatus}*/
+	/** Shurjopay status message. See more.. {@link ShurjopayStatus} */
```
Comment (imtiaz-rahi) [inline src/main/java/com/shurjomukhi/model/ShurjopayToken.java:49]: @JsonIgnore
	public String getFormatted() {
		return tokenType.concat(" ").concat(token);
	}

**Review by imtiaz-rahi — CHANGES_REQUESTED** [review]

File: src/main/java/com/shurjomukhi/model/ShurjopayToken.java (hunk 7 above)
Comment (a2alim) [reply src/main/java/com/shurjomukhi/model/ShurjopayToken.java:49]: applied this changes

File: src/main/java/com/shurjomukhi/Shurjopay.java (hunk 6 above)
Comment (a2alim) [reply src/main/java/com/shurjomukhi/Shurjopay.java]: applied this changes

**Review by imtiaz-rahi — APPROVED** [review]
looks ok

**Pull request merged**